modules/compiler/spirv-ll/include/spirv-ll/builder_debug_info.h

```diff
@@ -18,22 +18,135 @@
 #define SPIRV_LL_SPV_BUILDER_DEBUG_INFO_H_INCLUDED
 
 #include <spirv-ll/builder.h>
+#include <spirv-ll/module.h>
+#include <spirv-ll/opcodes.h>
 
 namespace spirv_ll {
 
 /// @brief Combined builder for the DebugInfo and OpenCLDebugInfo100 extended
 /// instruction sets.
 class DebugInfoBuilder : public ExtInstSetHandler {
  public:
+  enum Workarounds {
+    // Some versions of llvm-spirv mistakenly swap
+    // DebugTypeTemplateTemplateParameter and DebugTypeTemplateParameterPack
+    // opcodes, leading to incorrect binaries. When this workaround is enabled,
+    // we assume binaries have been created with this bug.
+    // See https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2248
+    TemplateTemplateSwapParamterPack = 1 << 0,
```

Comment:
Typo in the name. As long as the name needs updating anyway, something like `TemplateTemplateSwappedWithParameterPack` might be slightly clearer.

Reply:
Good spot (I rushed that bit, sorry) and yeah I've taken your suggestion on the renaming.

Furthermore, the workaround now tries its best to infer which instruction it's received. We have tests compiled with `spirv-as` without this bug and real-world binaries with `llvm-spirv` which do have this bug, so it's unreasonable to statically choose one or the other. It's not a foolproof system, but it should be fairly accurate in its determinations.